teps/0154-concise-remote-resolver-syntax.md

```diff
@@ -0,0 +1,290 @@
+* Define the formats for the existing Tekton's resolvers.
+
+### Non-Goals
+
+### Use Cases
+
+* As a task/pipeline author, I would like to reduce the verbosity of the yaml.
+
+### Requirements
+
+* The new resolver-reference should not be backwards incompatible. It should be an additive change.
+* The new resolver-reference should be optional. 
+
+## Proposal
+
+### Extending the `resolver` field
+
+Currently, when using a remote resolver, we need to provide the `resolver`. Furthermore, the `resolver`'s value needs to be a [`a lowercase RFC 1123 subdomain`](https://datatracker.ietf.org/doc/html/rfc1123) name. 
+
+```yaml
+taskRef:
+  resolver: <resolver-name>
+``` 
+
+We propose updating this requirement and checking that the value of the `resolver` can be of format `subdomain` or the Resolver's defined resolver-reference format. 
+
+### Resolver-Reference Format
+
+[Package-URL](https://github.com/package-url/purl-spec) is a format that we have begun to use in Tekton Artifacts. To enable a consistent concise format for across resolvers, we propose using purl.
+
+To provide a consistent format across resolvers, we suggest the following `remote-resolver` template: `<resolver-name>:<type>/<namespace>/<name>@<version>#<subpath>?<additional-params>`.
+
+The `Remote Resolver` must define the format using the above `remote-resolver` template. The `remote-resolver` would be responsible for parsing the `resolver-reference` and extracting the necessary information. Tekton will simply validate that it is a valid `purl` spec. The requirements for the fields in the template are defined in the [purl](https://github.com/package-url/purl-spec/blob/master/PURL-SPECIFICATION.rst) spec.
+
+If users provide an [invalid purl string](https://github.com/package-url/purl-spec/blob/master/PURL-SPECIFICATION.rst#rules-for-each-purl-component), a validation error will be thrown in the `TaskRun/PipelineRun's` `Status`.
+
+#### Git Resolver
+Based on `git` resolver's [params](https://tekton.dev/docs/pipelines/git-resolver/#parameters), we define the following purl spec:
+
+```
+git:<git-provider>/<org>/<repo>@<revision>#<pathInRepo>?token=<token>&tokenKey=<tokenKey>&serverURL=<serverURL>&scmType=<scmType>
+```
+
+Example:
+
+```
+git:github.com/tektoncd/catalog@sha256:053a6cb9f3711d4527dd0d37ac610e8727ec0288a898d5dfbd79b25bcaa29828#task/hello-world?token=git-token
+```
+
+Using the above example, a user can do the following when referencing the Task:
+```yaml
+taskRef:
+  resolver: "git:github.com/tektoncd/catalog@sha256:053a6cb9f3711d4527dd0d37ac610e8727ec0288a898d5dfbd79b25bcaa29828#task/hello-world?token=git-token"
+```
+
+#### Bundle Resolver
+Based on `bundle` resolver's [params](https://tekton.dev/docs/pipelines/bundle-resolver/#parameters), we define the following purl spec:
+
+```
+bundle:<bundle-provider>/<bundle-namespace>/<bundle-name>@<bundle-version>?kind=<kind>&name=<name>&secret=<secret>
```

Comment:
This also gets messy for similar reasons as git. oci spec doesn't make a restriction on path lengths - this can be 1...many parts long.

Reply:
Agreed!  I think its fixed now.